### PR TITLE
Extract metadata from song

### PR DESCRIPTION
**Pull request by krishrustagi**

### CODE REVIEW

File: src/main/java/com/emotunes/emotunes/dao/UserSongMappingDao.java [hunk 1]
```diff
@@ -82,7 +82,7 @@ public void songLiked(String userId, SongMetadata songMetadata, boolean isLiked)
     public void addSongsForUser(String userId) {
         List<StoredSong> songList = songRepository.findAll();
         songList.forEach(song -> {
-            Emotion songEmotion = Emotion.HAPPY; // todo: predict song emotion
+            Emotion songEmotion = Emotion.HAPPY; // use pre-defined emotions the song
```
Comment (SarthakSavasil) [inline src/main/java/com/emotunes/emotunes/dao/UserSongMappingDao.java]: Add todo in comment

File: src/main/java/com/emotunes/emotunes/entity/StoredSong.java [hunk 2]
```diff
@@ -26,5 +28,5 @@ public class StoredSong extends BaseEntity {
 
     private LocalTime duration;
 
-    // todo: add thumbnail
```
Comment (SarthakSavasil) [inline src/main/java/com/emotunes/emotunes/entity/StoredSong.java:29]: Solved? How?

Comment (SarthakSavasil) [reply src/main/java/com/emotunes/emotunes/entity/StoredSong.java:29]: Add songUrl and thumbnailUrl to DB

File: src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java [hunk 3]
```diff
@@ -42,21 +48,37 @@ public ResponseEntity<String> addSong(MultipartFile songFile) throws IOException
...
                             ).toLocalTime().toString())
+                            .artist(artist)
                             .build();
 
             String songId = persistSong(songMetadata);
 
+            Artwork artwork = tag.getFirstArtwork();
+            if (artwork != null) {
+                byte[] imageData = artwork.getBinaryData();
+                File thumbnail = new File(songId + ".jpg");
```
Comment (SarthakSavasil) [inline src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: isko file object ko close nhi krna hota kya?

File: src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java [hunk 4]
```diff
@@ -42,21 +48,37 @@ public ResponseEntity<String> addSong(MultipartFile songFile) throws IOException
```
Comment (SarthakSavasil) [inline src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java:73]: Abhi ke liye theek hai But insert todo to use multithreading

File: src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java [hunk 5]
```diff
@@ -42,21 +48,37 @@ public ResponseEntity<String> addSong(MultipartFile songFile) throws IOException
...
+                byte[] imageData = artwork.getBinaryData();
+                File thumbnail = new File(songId + ".jpg");
+                ByteArrayInputStream inputStream = new ByteArrayInputStream(imageData);
+                BufferedImage bufferedImage = ImageIO.read(inputStream);
+                ImageIO.write(bufferedImage, "jpg", thumbnail);
+
+                // save thumbnail file
+            }
+
             List<StoredUser> userList = userDao.findAll();
```
Comment (SarthakSavasil) [inline src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: Split it into a function call

File: src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java (hunk 3 above)
Comment (SarthakSavasil) [reply src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: Split it into a function

File: src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java (hunk 4 above)
Comment (krishrustagi) [reply src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java:73]: doosri pr mein daal dunga

File: src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java [hunk 6]
```diff
@@ -38,53 +45,56 @@ public class AdminServiceImpl implements AdminService {
...
 
         try {
             AudioFile audioFile = AudioFileIO.read(file);
-            long duration = audioFile.getAudioHeader().getTrackLength();
-            log.info("{}", Instant.ofEpochSecond(duration).atZone(
-                    ZoneId.of("UTC")
-            ).toLocalTime().toString());
+            Tag tag = audioFile.getTag();
+            String title = tag.getFirst(FieldKey.TITLE);
+            String artist = tag.getFirst(FieldKey.ARTIST);
```
Comment (SarthakSavasil) [inline src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: if Artist name not present what will it contain?

Comment (SarthakSavasil) [reply src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: If null add a null check and pass "<Unknown>" as artist

**Review by SarthakSavasil — APPROVED** [review]

**Pull request merged**